### ISSUE TITLE
Can not set data to session

### ISSUE THREAD

**Issue by kasperdun** [crisp-im/crisp-sdk-android#198]
I can't set data to session
I tried set data before and after chat activity showed
I tried resetChatSession and not
I tried setTokenID and not

Then I do Crisp.setSessionInt("currentUser", user.id) and currentUser is not shown on other side

Help me understand were I'm wrong

```kt
// in my main activity onCreate
Crisp.configure(applicationContext, "websiteID")

// in my view

@Composable
fun HelpMenuScreen(
    modifier: Modifier = Modifier,
    onGoToTickets: () -> Unit,
) {
    val viewModel: HelpMenuViewModel = hiltViewModel()
    val state by viewModel.uiState.collectAsStateWithLifecycle()

    val context = LocalContext.current

    with(state) {
        HelpMenuScreenContent(
            ...
            onOpenSupportChat = {

                val startCrispActivity = {
                    val crispIntent = Intent(
                        context,
                        ChatActivity::class.java
                    )
                    context.startActivity(crispIntent)
                }

                viewModel.setCrispSettingWithAction(
                    context = context,
                    action = startCrispActivity
                )
            }
        )
    }
}

// in my view model setting crisp session and call action callback to show chat activity 
fun setCrispSettingWithAction(
        context: Context,
        action: () -> Unit,
    ) {
        viewModelScope.launch {
            try {
                val data = profileRuntimeStore.getStateFlow().firstOrNull() ?:  throw Throwable()
                ...
                data.userDetails?.also { user ->
                    Crisp.setTokenID(context, user.id.toString())
                    user.email.also(Crisp::setUserEmail) // works
                    
                    Crisp.setSessionInt("currentUser", user.id) // not works
                }
                ...

                action()

            } catch (e: Exception) {
                alertNotificationManager.onEmitServerErrorEvent()
            }
        }
    }
```

**Comment by Doc1faux** [comment]
Hi @kasperdun,

Which version of the SDK are you using?
`Crisp.setSessionInt` works well on my side with your workflow on latest `v2.0.8`.
```java
Crisp.configure(appContext, "websiteID")
Crisp.setTokenID(context, user.id.toString())
Crisp.setUserEmail(user.email)
Crisp.setSessionInt("currentUser", user.id)
val context = LocalContext.current
val crispIntent = Intent(context, ChatActivity::class.java)
context.startActivity(crispIntent)
```

Keep in mind that, since [`v2.0.2beta3`](https://github.com/crisp-im/crisp-sdk-android/releases/tag/v2.0.2beta3), we had to add a debounce of 6s to batch session [data](https://github.com/crisp-im/crisp-sdk-android/wiki/2.-Available-APIs#push-session-data) & [segments](https://github.com/crisp-im/crisp-sdk-android/wiki/2.-Available-APIs#push-session-segments) events to avoid being rate limited by the API.

However, I've just seen that if the timer reaches the 6s after the chatbox is closed, `data` & `segments` are lost if set...
I've already found the fix and will push it along with other updates on notifications.

**Comment by Doc1faux** [comment]
@kasperdun Could you just confirm me that if you keep your chatbox opened a little more than 6s, your `currentUser` int is correctly sent and visible on the dashboard?

**Comment by kasperdun** [comment]
Hi @Doc1faux,
I use 2.0.8 too, and I think problem with 6s debounce not my problem, but I glad to hear you found some problem and can fix it

When I show chat and I type message, type my name, and waiting answer from our support about visibility of currentUser and others data in session - it takes more then 6s, I think about 1m

I have iOS app and use Crisp sdk too, data in session visible in my case for iOS

**Comment by kasperdun** [comment]
Yes, here is my log
I see 2 session:set_data
but I have 4 setSessionString, 2 setSessionInt and 1 setSessionBool in code before show chat
and no data got in dashboard

```yaml

Disconnecting…
Adding listener. Number of listeners is 1.
Starting chat initialization timeout of 15 seconds.
Connecting…
Connected to WebSocket.
SESSION CACHE - WEBSOCKET CONNECT
Found saved session.
Sending action session:join
Clearing chat initialization timeout.
Scheduling heartbeat timer…
Received action session:joined
Sending action notification:provider:register
Sending action session:set_email
Sending action storage:sync:update
Sending action session:set_nickname
Sending action helpdesk:article:search
Received action storage:sync:updated
Received action helpdesk:article:searched
Sending action notification:provider:register
Sending action session:set_data
Sending action session:set_data
```

**Comment by Doc1faux** [comment]
There should only be one `session:data`, due to the 6s debounce timer, we merge any `session:set_data` you call (either `Crisp.setSessionInt`, `Crisp.setSessionBool` or `Crisp.setSessionString`). I found why there is a duplicate, it will also be fixed, but it has currently no impact on your issue.
The same can occur on `session:set_segments` and it will be fixed too.

As you, I called 4 setSessionString, 2 setSessionInt and 1 setSessionBool before opening the chatbox, the message is correctly sent through the WebSocket, but... not received in the dashboard?!?

Nice finding! I'll dive into it!

**Comment by Doc1faux** [comment]
WoW, I just found why mine were not visible in the dashboard!

As soon as I removed spaces in my keys, all was being visible, so maybe you have some illegal characters in your keys.
Could you share them with me?

If you're not confident sharing them here, could you send us a message through our chat at [Crisp](https://crisp.chat/fr/), linking this issue and asking to talk to me?

**Comment by Doc1faux** [comment]
Your other issue is explained [here](https://github.com/crisp-im/crisp-sdk-android/wiki/2.-Available-APIs#reset-session).

> Keep in mind that if you have set a tokenID in order to create this session, it will not be reset until you call [Crisp.setTokenID(Context, String)](https://github.com/crisp-im/crisp-sdk-android/wiki/2.-Available-APIs#session-continuity), so starting the ChatActivity will restore the session.

**Comment by Doc1faux** [comment]
I think I've found the regex which check for the `key` validity in the minified Web SDK: `/^([A-Za-z0-9_\\-]{1,50})$`

**Comment by kasperdun** [comment]
All my keys valid with this regex, they not included spaces or any illegal characters.
All of them something like "userEmail"
I send code to your crisp chat

Answer: 
> I sent all the details to the tech team and as soon as they give me a feedback I'll let you know.

**Comment by Doc1faux** [comment]
I've seen the conversation, could you provide us also a sample of what you send also into values?
You can make them up from a dataset you really send, just keep the format you sent them with.
I am wondering if there is any character which got filtered by the backend and the whole request ignored.

**Comment by Doc1faux** [comment]
It seems you try to send JSON into string session data, it was not designed for this, but it seems to work.

However, each data has a 200 chars limit and your stringified JSON hit it (~300 chars), that's why the server is rejecting the whole request ;)

Our Backend developer is suggesting you to split your json into multiple data entries.

As this limit will not be increased, I close the issue.

For info, I already fixed the issue below which I raised in [this comment](https://github.com/crisp-im/crisp-sdk-android/issues/198#issuecomment-2536308531) and it 'll be available in the next release.

>However, I've just seen that if the timer reaches the 6s after the chatbox is closed, data & segments are lost if set...
I've already found the fix and will push it along with other updates on notifications.
